[RPC] Add masternode's pubkey to listmasternodes RPC

presstab opened the pull request:
Add the masternode's public key (used to broadcast masternode messages) to `listmasternodes` RPC. I could not find another convenient way to get a masternode's public key (which is public information needed to validate masternode messages) from the RPC calls.

random-zebra commented:
The `GetHex` method of CPubKey seems indeed off, or, at least, its name is misleading as it doesn't even return hex data.
With the fix in https://github.com/PIVX-Project/PIVX/pull/743 (basically the same as `HexStr`... don't know performance wise) the output of this  would be:
```
{
    "rank": 45,
    "network": "ipv4",
    "txhash": "df56577ddbf104f5e780d9a92d3386ddcae618e139a4926b0f8a71442778e69e",
    "outidx": 0,
    "pubkey": "047a852480a76e325d47221e9de54ac3a2cd148838fbac75124daa70afcdb9d0afa5347babd5d9112e5cb0aaab3d50a34b0737e534725c15861cab2af7f84a407c",
    "status": "ENABLED",
    "addr": "yJYqCR8kV71CtUKgUwhY2FVrtquc4ESwWy",
    "version": 70914,
    "lastseen": 1539219823,
    "activetime": 260007,
    "lastpaid": 1539215749
  }
```

but maybe for the purpose of this RPC call it's sufficient to return the pubkey as Base58check encoded string (as for the collateral key) e.g.:
`
obj.push_back(Pair("pubkey", CBitcoinAddress(mn->pubKeyMasternode.GetID()).ToString()));
`

giving (for the same node)
```
 {
    "rank": 27,
    "network": "ipv4",
    "txhash": "df56577ddbf104f5e780d9a92d3386ddcae618e139a4926b0f8a71442778e69e",
    "outidx": 0,
    "pubkey": "yGEfBx65h1ypqVEybzgs4K52uKdC1okdU5",
    "status": "ENABLED",
    "addr": "yJYqCR8kV71CtUKgUwhY2FVrtquc4ESwWy",
    "version": 70914,
    "lastseen": 1539219823,
    "activetime": 260007,
    "lastpaid": 1539215749
  }
```

which would be consistent with the output of other calls like `decodemasternodebroadcast`.

Fuzzbawls commented on src/rpc/masternode.cpp:
```diff
@@ -252,6 +253,7 @@ UniValue listmasternodes(const UniValue& params, bool fHelp)
             obj.push_back(Pair("network", strNetwork));
             obj.push_back(Pair("txhash", strTxHash));
             obj.push_back(Pair("outidx", (uint64_t)oIdx));
+            obj.push_back(Pair("pubkey", mn->pubKeyMasternode.GetHex()));
```
As previously mentioned, looks like using `.GetHex()` is producing some ugly output.

recommend changing to
```c++
obj.push_back(Pair("pubkey", HexStr(mn->pubKeyMasternode)));
```
or, alternatively the format suggested by @random-zebra, whichever format you decide on using for zDEX.

presstab commented:
Thanks for the quick fix @Mrs-X 

```
  {
    "rank": 1336,
    "network": "onion",
    "txhash": "c2bae5c4a73e8e9e1a259038af3f8670ff045898f8ad98a23714be8547bc5c92",
    "outidx": 0,
    "pubkey": "047948174478d9f07edcda508b55bbea500d82795130d13deea40bad59320731cbc0ffdd4f7701c8126e7cbeeaee0765e5ef35184174389999a9b6a8a58a29a2ca",
    "status": "ENABLED",
    "addr": "DFcCbyTSAdJvVvYvvRfva1D22t2boGZBTr",
    "version": 70914,
    "lastseen": 1539277559,
    "activetime": 8769581,
    "lastpaid": 1539226840
  },
```

Fixed and squashed onto previous commit.
